mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -48,305 +48,410 @@
+        }
 
-        Preconditions.checkNotNull(message.getMailboxId());
-        Preconditions.checkArgument(!users.isEmpty());
-        IndexableMessage indexableMessage = new IndexableMessage();
-        try {
+        private IndexableMessage instanciateIndexedMessage() throws IOException, MimeException {
+            String messageId = message.getMessageId().serialize();
             MimePart parsingResult = new MimePartParser(message, textExtractor).parse();
-            indexableMessage.users = users.stream().map(User::getUserName).collect(Guavate.toImmutableList());
-            indexableMessage.bodyText = parsingResult.locateFirstTextBody();
-            indexableMessage.bodyHtml = parsingResult.locateFirstHtmlBody();
-            indexableMessage.hasAttachment = message.getProperties()
-                .stream()
-                .anyMatch(property -> property.equals(HAS_ATTACHMENT_PROPERTY));
-            indexableMessage.setFlattenedAttachments(parsingResult, indexAttachments);
-            indexableMessage.copyHeaderFields(parsingResult.getHeaderCollection(), getSanitizedInternalDate(message, zoneId));
-            indexableMessage.generateText();
-        } catch (IOException | MimeException e) {
-            throw Throwables.propagate(e);
+
+            List<String> stringifiedUsers = users.stream().map(User::getUserName).collect(Guavate.toImmutableList());
+
+            Optional<String> bodyText = parsingResult.locateFirstTextBody();
+            Optional<String> bodyHtml = parsingResult.locateFirstHtmlBody();
+
+            boolean hasAttachment = computeHasAttachment(message);
+            List<MimePart> attachments = setFlattenedAttachments(parsingResult, indexAttachments);
+
+            HeaderCollection headerCollection = parsingResult.getHeaderCollection();
+            ZonedDateTime internalDate = getSanitizedInternalDate(message, zoneId);
+
+            Multimap<String, String> headers = headerCollection.getHeaders();
+            Subjects subjects = Subjects.from(headerCollection.getSubjectSet());
+            EMailers from = EMailers.from(headerCollection.getFromAddressSet());
+            EMailers to = EMailers.from(headerCollection.getToAddressSet());
+            EMailers replyTo = EMailers.from(headerCollection.getReplyToAddressSet());
+            EMailers cc = EMailers.from(headerCollection.getCcAddressSet());
+            EMailers bcc = EMailers.from(headerCollection.getBccAddressSet());
+            String sentDate = DateResolutionFormater.DATE_TIME_FOMATTER.format(headerCollection.getSentDate().orElse(internalDate));
+
+            String text = Stream.of(from.serialize(),
+                    to.serialize(),
+                    cc.serialize(),
+                    bcc.serialize(),
+                    subjects.serialize(),
+                    bodyText.orElse(null),
+                    bodyHtml.orElse(null))
+                    .filter(str -> !Strings.isNullOrEmpty(str))
```

Comment:
can you differenciate indentation between stream initialization and method calls?